File: frontend/src/shared/types/media.ts
export type MediaType = 'film' | 'series';

export interface Film {
  id: number;
  title: string;
  genre: string;
  company: string;
  year: number;
  rating: number;
  type: MediaType;
}

export interface Series extends Film {
  seasons: number;
  episodes: number;
}

export type MediaItem = Film | Series;

File: backend/internal/models/media.go
package models

type Media interface {
	GetID() int
	GetTitle() string
	GetGenre() string
	GetRating() float64
	GetType() string
}

type MediaStruct struct {
	ID       int     `json:"id"`
	Title    string  `json:"title"`
	Genre    string  `json:"genre"`
	Company  string  `json:"company"`
	Year     int     `json:"year"`
	Rating   float64 `json:"rating"`
	Seasons  *int    `json:"seasons"`
	Episodes *int    `json:"episodes"`
	Type     string  `json:"type"`
}